test/README.md

```diff
@@ -0,0 +1,154 @@
+
+> Feel free to suggest improvements to the template, there is still a lot of room for improvement. 
+
+We recommend the following approach when you create a unit test:
+
+* For new utilities, we recommend creating a new spec.
+* For existing utilities, if the method is already being tested, just add a new `test()` block.
+* Notice that all API spec files are appended with `api.[feature].spec.js`, we recommend to follow the same approach. eg: `api.[deprecate].spec.js`.
+* Don't mix utilities with API tests.
+
+### Functional tests
+
+The functional tests aim to run only **cli endpoint** and **web point** using real request to an existing and compiled running Verdaccio server.
+
+> Be aware if you change something in the `{root}/src` source code, you must run `yarn code:build` before to be able to see your changes because functional tests use the transpiled code. 
+
+All tests must be included in the `test/functional/index.spec.js` file, which bootstraps the whole process. There is only one spec file and **must be only one**.
+
+The jest configuration file is defined in `test/jest.config.functional.js`. The configuration will create a custom environment launching 3 Verdaccio servers with different configurations.
+
+The servers are linked as follows: 
+
+* Server 1 
+  * -> Server 2  
+  * -> Server 3
+* Server 2 
+  * -> Server 1
+* Server 3 
+  * -> Server 2
+  * -> Server 1
+* Express app: (if you need to emulate any external endpoint, use the express app)  
+
+Server 1 runs on port `55551`, Server 2 on port `55552` and Server 3 on port `55553`.
+
+> If you have the need to increase the number of servers running, it is possible, but please discuss with the team before you go in that path.
+
+
+#### Adding a new block
+
+To add a new feature you need to export the feature as a function that take as an argument any of the servers you want to interact. 
+
+
+```js
+// newFeature.js
+
+export default function(server) {
+
+  describe('package access control', () => {
+    test('should ...', (done) => {
+      done();
+    });
+  });
+  
+}
+```
+
+Then, import the feature and run the function within the main `describe` block.  
+
+```js
+// index.spec.js
+
+import newFeature from './newFeature';
+
+describe('functional test verdaccio', function() {
+  // test are fast, but do not change this time out, 10 seconds should be more than enough
+  jest.setTimeout(10000);
+  // servers are accessed vi a global jest state.
+  const server1: IServerBridge = global.__SERVERS__[0];
+  const server2: IServerBridge = global.__SERVERS__[1];
+  const server3: IServerBridge = global.__SERVERS__[2];
+  const app = global.__WEB_SERVER__.app;
+
+  // include as much servers you need
+  newFeature(server1, server2, server3);
+});
+```
+
+Functional test run over ones single file, thus, it is not possible at this stage to run test individually.
```

Comment:
```suggestion
Functional tests run over one single file, thus, it is not possible at this stage to run tests individually.
```

Reply:
it's possible however to use jest's `test.only` or `test.skip` and using that to selectively run or ignore specific tests.

Reply:
Actually, you are right